float_range/range.py

```diff
@@ -63,11 +64,27 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __len__(self):
-        if self._is_empty():
+        raw_len = abs((self.major - self.minor) / self.step)
+
+        if self._is_empty() or raw_len < 1:
             return 0
 
-        raw_len = abs(self.major - self.minor)
-        return int(raw_len/self.step)
+        return int(math.ceil(raw_len))
+
+    def __getitem__(self, key):
+        if not isinstance(key, int):
+            raise TypeError
+
+        pos_item = self.step * key + self.start
+        neg_item = self.step * (len(self) + key) + self.start
+
+        if pos_item in self:
+            return pos_item
+
+        if neg_item in self:
+            return neg_item
+
+        raise IndexError
```

Comment:
Same thing here. Perhaps we could make something like `range`:
``` python
>>> x = range(10)
>>> x[-1]
9
>>> x[-2]
8
>>> x[10]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: range object index out of range
```
So something like:
```python
class_name = self.__class__.__name__
raise IndexError('{} object index out of range'.format(class_name))
```